docs/jwst/data_products/msa_metadata.rst

```diff
@@ -0,0 +1,221 @@
+.. _msa_metadata:
+
+MSA Meta Data File: ``msa``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+While not containing any actual science data, the NIRSpec MSA meta data file is nonetheless
+a crucial component of calibration processing for NIRSpec MOS exposures.
+It contains all the slitlet, shutter, and source configuration information that's needed
+by the :ref:`calwebb_spec2 <calwebb_spec2>` pipeline to process a MOS exposure.
+These meta data are originally generated by the
+`MSA Planning Tool <https://jwst-docs.stsci.edu/display/JDOX/.NIRSpec+MSA+Planning+Tool%2C+MPT+v2.0>`_
+(MPT) in APT and passed through the ground system to populate the
+FITS metafile. Details of the meta data and how it is used in calibration processing
+are given below.
+
+File names for MSA meta data files are of the form:
+
+ jw<ppppp><ooo><vvv>_<cfg>_msa.fits
+
+where
+
+ - ppppp: program ID number
+ - ooo: observation number
+ - vvv: visit number
+ - cfg: MSA configuration
+
+such as `jw01180025001_01_msa.fits`.
+
+Each MOS science exposure contains a primary header keyword `MSAMETFL` that gives
+the name of the MSA meta data file to be used when processing the exposure.
+The values of the primary header keywords `MSAMETID` and `PATT_NUM` are also used
+when retrieving relevant rows of data from the tables in the MSA meta data file.
+If any of these keywords is missing from the input science exposure, processing
+will not be able to proceed and an error will be raised.
+
+File Structure
+--------------
+The overall structure of the MSA FITS file is as follows:
+
++-----+---------------+----------+-----------+--------------------+
+| HDU | EXTNAME       | HDU Type | Data Type | Dimensions         |
++=====+===============+==========+===========+====================+
+|  0  | N/A           | primary  | N/A       | N/A                |
++-----+---------------+----------+-----------+--------------------+
+|  1  | SHUTTER_IMAGE | IMAGE    | float32   | 342 x 730          |
++-----+---------------+----------+-----------+--------------------+
+|  2  | SHUTTER_INFO  | BINTABLE | N/A       | variable x 12 cols |
++-----+---------------+----------+-----------+--------------------+
+|  3  | SOURCE_INFO   | BINTABLE | N/A       | variable x 8 cols  |
++-----+---------------+----------+-----------+--------------------+
+
+ - PRI: The primary data array is empty. The primary header contains a small amount of
+   meta data related to the observation and MSA configuration.
+ - SHUTTER_IMAGE: 2-D data array containing a map of MSA shutters that've been commanded
```

Comment:
If these are only the ones that have been commanded open, how does the pipeline find out about shutters that are permanently open or permanently closed?